### PR TITLE
Improve batch job submit validation efficiency

### LINKED ISSUE #2178: Submitting a batch job with file_path_lister is failing/timing out
Try the following config:

```
{
    ...
    "file_path_lister": {
        "s3_paths": [<string>], 
    }
}
```

When a list of 600 files is passed, the request times out when making a POST request using python's request library.

### PR DESCRIPTION
**Pull request by deliahu**
closes #2178

### CODE REVIEW

**Review by miguelvr — COMMENTED** [review]
Can we also take this opportunity and merge the types / functions? 

I think they can exist in the enqueuer package and be imported by the operator

**Comment by deliahu** [comment]
> Can we also take this opportunity and merge the types / functions?
> 
> I think they can exist in the enqueuer package and be imported by the operator

As discussed live, we will address this in a separate PR

File: pkg/operator/schema/job_submission.go [hunk 1]
```diff
@@ -28,9 +28,10 @@ type ItemList struct {
 }
 
 type S3Lister struct {
-	S3Paths  []string `json:"s3_paths"` // s3://<bucket_name>/key
-	Includes []string `json:"includes"`
-	Excludes []string `json:"excludes"`
+	S3Paths    []string `json:"s3_paths"` // s3://<bucket_name>/key
+	Includes   []string `json:"includes"`
+	Excludes   []string `json:"excludes"`
+	MaxResults *int64   `json:"max_results"` // this is not currently exposed to the user (it's used for validations)
```
Comment (miguelvr) [inline pkg/operator/schema/job_submission.go:34]: if it is not exposed, should have a json tag like so: `json:"-"`

File: pkg/enqueuer/enqueuer.go [hunk 2]
```diff
@@ -56,9 +56,10 @@ type ItemList struct {
 }
 
 type S3Lister struct {
-	S3Paths  []string `json:"s3_paths"` // s3://<bucket_name>/key
-	Includes []string `json:"includes"`
-	Excludes []string `json:"excludes"`
+	S3Paths    []string `json:"s3_paths"` // s3://<bucket_name>/key
+	Includes   []string `json:"includes"`
+	Excludes   []string `json:"excludes"`
+	MaxResults *int64   `json:"max_results"` // this is not currently exposed to the user (it's used for validations)
```
Comment (miguelvr) [inline pkg/enqueuer/enqueuer.go:62]: if it is not exposed, should have a json tag like so: json:"-"

**Review by miguelvr — APPROVED** [review]

**Pull request merged**